### PR TITLE
HPCC-16550 Ensure __COMPRESSED__ flag applied to external files

### PR DESCRIPTION
**Pull request by jakesmith**
Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

### CODE REVIEW

**Comment by jakesmith** [comment]
@ghalliday - please review

File: thorlcr/activities/thdiskbase.cpp [hunk 1]
```diff
@@ -48,6 +48,10 @@ void CDiskReadMasterBase::init()
     Owned<IDistributedFile> file = queryThorFileManager().lookup(container.queryJob(), helperFileName, 0 != ((TDXtemporary|TDXjobtemp) & helper->getFlags()), 0 != (TDRoptional & helper->getFlags()), true);
     if (file)
     {
+        CDfsLogicalFileName lfn;
```
Comment (ghalliday) [inline thorlcr/activities/thdiskbase.cpp]: Would it be worth adding a file->isExternal() call?  It would save processing the filename again.

**Comment by ghalliday** [comment]
@jakesmith the code looks fine - happy to merge as-is.  One question about the implementation.

**Comment by jakesmith** [comment]
@ghalliday - have added a isExternal() method to IDistributedFile.

There's a few more changes because I also removed a unused parameter from constructor that was in same area.

File: dali/base/dadfs.cpp [hunk 2]
```diff
@@ -2679,6 +2679,7 @@ class CDistributedFileBase : implements INTERFACE, public CInterface
     unsigned defaultTimeout;
     bool dirty;
     Owned<IRemoteConnection> superOwnerLock;
+    bool external = false;
```
Comment (ghalliday) [inline dali/base/dadfs.cpp]: minor: moving this field before superOwnerLock would save 8 bytes of memory per object.

**Comment by ghalliday** [comment]
@jakesmith looks good, 1 minor comment

**Comment by jakesmith** [comment]
@ghalliday - have moved bool (Wouldn't normally even thing about it , except in objects that I expect a lot of )

**Pull request merged**